c/tskit/tables.c

```diff
@@ -7799,6 +7799,317 @@ tsk_table_collection_clear(tsk_table_collection_t *self)
     return tsk_table_collection_truncate(self, &start);
 }
 
+static int TSK_WARN_UNUSED
+node_table_copy_row(tsk_node_table_t *source, tsk_node_table_t *dest, tsk_id_t row,
+    tsk_id_t *population_map, tsk_id_t *individual_map)
+{
+    int ret = 0;
+    tsk_id_t pop, ind;
+    tsk_size_t metadata_length;
```

Comment:
Actually, it'd save a bit of trouble if you did
```c
ret = node_table_get_row(source, row, &node);
if (ret < 0) {
    goto out;
}
```
This will do the bounds checking for you, and save some tedium in dealing with the offsets.